crates/yamlpatch/src/lib.rs

```diff
@@ -209,42 +209,13 @@ pub fn apply_yaml_patches(
     Ok(next_document)
 }
 
-/// Compute the replacement range for a byte span, including any trailing newline
-/// that exists immediately after the span in the document.
-fn compute_replacement_range_for_span(
-    byte_span: (usize, usize),
-    document: &yamlpath::Document,
-) -> std::ops::Range<usize> {
-    let start = byte_span.0;
-    let mut end = byte_span.1;
-    let content = document.source();
-
-    // Only include trailing newline if it's the last character in the document.
-    // This preserves document-level trailing newlines without interfering with
-    // newlines that are part of the document structure.
-    if end + 1 == content.len() && content.as_bytes()[end] == b'\n' {
-        end += 1;
-    }
```

Comment:
I didn't try to prove it out, but I think the reason this didn't work is because some incoming patches _didn't_ have a trailing newline, and so if `tree-sitter-yaml` didn't include one in the captured node for whatever reason we'd end up trimming it.